op-e2e/blobs/blob_test.go

```diff
@@ -0,0 +1,178 @@
+package blobs
+
+import (
+	"bytes"
+	"context"
+	"crypto/ecdsa"
+	"errors"
+	"fmt"
+	"math/big"
+	mrand "math/rand"
+	"testing"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-e2e"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/wait"
+	"github.com/ethereum-optimism/optimism/op-e2e/system/e2esys"
+	"github.com/ethereum-optimism/optimism/op-node/node"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/testutils"
+	"github.com/ethereum-optimism/optimism/op-service/txmgr"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/consensus/misc/eip4844"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethstorage/da-server/pkg/da"
+	"github.com/ethstorage/da-server/pkg/da/client"
+	"github.com/holiman/uint256"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	dacPort = 37777
+)
+
+var (
+	ctx       = context.Background()
+	dacUrl    = fmt.Sprintf("http://127.0.0.1:%d", dacPort)
+	toAddress = testutils.RandomAddress(mrand.New(mrand.NewSource(dacPort)))
+	blobs     = make([]*eth.Blob, 3)
+)
+
+func TestFunctionSuccess(t *testing.T) {
+	op_e2e.InitParallel(t)
+	dacServer := StartDACServer(t)
+	defer dacServer.Stop(ctx)
+
+	sys, l2Client := StartSystemWithDAC(t)
+	t.Cleanup(sys.Close)
+
+	for i := range blobs {
+		b := GetRandBlob(t, int64(i))
+		blobs[i] = &b
+	}
+
+	tx, err := sendTransactionWithBlobs(t, ctx, l2Client, sys.TestAccount(0), toAddress, blobs)
+	require.NoError(t, err)
+	_, err = wait.ForReceiptOK(ctx, l2Client, tx.Hash())
+
+	dblobs, err := downloadBlobs(dacUrl, tx.BlobHashes())
+	require.NoError(t, err)
+	require.True(t, len(dblobs) == len(tx.BlobHashes()), "blobs downloaded is not equal to blob hashes")
+
+	for i, blobHash := range dblobs {
+		blob := dblobs[i]
+		if len(blob) != eth.BlobSize {
+			t.Error("Invalid downloaded blob len", "blob hash", blobHash, "blob len", len(blob))
+		}
+		if bytes.Compare(blob, blobs[i][:]) != 0 {
+			t.Error("blob content", blob[:32], blobs[i][:32])
+		}
+	}
+}
+
+func StartSystemWithDAC(t *testing.T) (*e2esys.System, *ethclient.Client) {
```

Comment:
Don't use capitalized first character unless you want to export it. The same for other functions.